internal/compiler-bridge/src/main/scala/xsbt/DelegatingReporter.scala

```diff
@@ -90,10 +90,7 @@ private object DelegatingReporter {
 
     def makePosition(pos: Position): xsbti.Position = {
       val src = pos.source
-      val sourcePath = src.file match {
-        case AbstractZincFile(virtualFile) => virtualFile.id
-        case af: AbstractFile              => af.path
-      }
+      val sourcePath = src.file.path
```

Comment:
Not all sources are meant to be physical files expressible by `java.io.File`, so I think it's fine for sourcePath to contain `virtualFile.id`. This is also because some tooling might to save the compiler warnings as a machine-independent way.

Reply:
`src.file` is of type `AbstractZincFile` which have 2 implementations, `ZincPlainFile` and `ZincVirtualFile`:
- `path` on a `ZincPlainFile` returns the physical file path
- `path` on a `ZincVirtualFile` returns the `virtualFile.id`

So this implementation is safe and makes `sourceFile` and `sourcePath` consistent. That being said, I can understand that you want to push machine-independent path forward. But the fix of https://github.com/sbt/sbt/issues/5853 would then be more involved.